incubator/hnc/api/v1alpha1/hnc_config.go

```diff
@@ -69,10 +69,11 @@ type TypeSynchronizationStatus struct {
 	// Kind to be configured.
 	Kind string `json:"kind,omitempty"`
 
-	// Tracks the number of original objects that are being propagated to descendant namespaces.
+	// Tracks the number of objects that are being propagated to descendant namespaces. The propagated
+	// objects are created by HNC.
 	// +kubebuilder:validation:Minimum=0
 	// +optional
-	NumPropagated *int32 `json:"numPropagated,omitempty"`
+	NumPropagatedObjects *int32 `json:"numPropagatedObjects,omitempty"`
```

Comment:
Why int32 and not just int?

Reply:
Changing this to "int" would let you get rid of a lot of casts in the rest of the code.

Reply:
I thought we cannot use `int` from the documentation [here](https://book.kubebuilder.io/cronjob-tutorial/api-design.html), but looks like we can. Changed to `int`.